src/test/java/org/jvnet/hudson/test/JenkinsRuleTest.java

```diff
@@ -127,6 +142,206 @@ public void testTokenHelperMethods() throws Exception {
+        assertEquals(response.getStatusCode(), 200);
+
+    }
+
+    @Test
+    public void postJSONTests() throws IOException {
+
+        User admin = User.getById("admin", true);
+        MockAuthorizationStrategy auth = new MockAuthorizationStrategy()
+                .grant(Jenkins.ADMINISTER).everywhere().to(admin);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(auth);
+
+        JenkinsRule.WebClient webClient = j.createWebClient();
+        JenkinsRule.JSONWebResponse response;
+
+        // Testing an authenticated POST that should answer 200 OK and return same json
+        MyJsonObject objectToSend = new MyJsonObject("Creating a new Object with Json.");
+        response = j.postJSON( "testing-cli/create", JSONObject.fromObject(objectToSend),
+                               webClient, admin);
+        assertTrue(response.getContentAsString().contains("Creating a new Object with Json. - CREATED"));
+        assertEquals(response.getStatusCode(), 200);
+
+        // Testing an authenticated POST that return error 500
+        webClient.setThrowExceptionOnFailingStatusCode(false);
+        response = j.postJSON( "testing-cli/createFailure", JSONObject.fromObject(objectToSend),
+                               webClient, admin);
+        assertTrue(response.getContentAsString().contains("Creating a new Object with Json. - NOT CREATED"));
+        assertEquals(response.getStatusCode(), 500);
+
+    }
+
+    @Test
+    public void putJSONTests() throws Exception {
+
+        JenkinsRule.WebClient webClient = j.createWebClient();
+        JenkinsRule.JSONWebResponse response;
+
+        // Testing a simple PUT that should answer 200 OK and return same json
+        MyJsonObject objectToSend = new MyJsonObject("Jenkins is the way !");
+        response = j.putJSON( "testing-cli/update", JSONObject.fromObject(objectToSend), webClient);
+        assertTrue(response.getContentAsString().contains("Jenkins is the way ! - UPDATED"));
+
+        //Testing with a PUT that the test expect to raise an server error: we want to be able to assert the status
+        webClient.setThrowExceptionOnFailingStatusCode(false);
+        response = j.putJSON( "testing-cli/updateFailure", JSONObject.fromObject(objectToSend), webClient);
+        assertEquals(response.getStatusCode(), 500);
+        assertTrue(response.getContentAsString().contains("Jenkins is the way ! - NOT UPDATED"));
+
+        //Testing a PUT that requires the user to be authenticated
+        User admin = User.getById("admin", true);
+        MockAuthorizationStrategy auth = new MockAuthorizationStrategy()
+                .grant(Jenkins.ADMINISTER).everywhere().to(admin);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(auth);
+
+        // - simple call without authentication should be forbidden
```

Comment:
(Note distinction between `RootAction` and `UnprotectedRootAction`.)

Reply:
yes to me the default should be protected (so RootAction) so I prefer to do example with this one.